### DISCUSSION TITLE
Rename the "Airstreams" contract to something more general-purpose

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#183] [Ideas]
It dawned on me that the name of the `Airstreams` contract need not be as specific as it currently is.

We should consider renaming the contract to something more general-purpose because the same contract could facilitate other use cases, such as non-airdrop vesting. The target audience is normal Sablier senders who want to stream to a large number of recipients; subsidizing the gas costs for everybody is not feasible because only up to ~100 streams can be created in a single Ethereum block.

Suggested alternative names:

- `MerkleStreamer` 
- `MerkleDistributor` (the name Uniswap used for its airdrop, see https://github.com/Uniswap/merkle-distributor)

To avoid any doubts, I don't suggest rebranding the Airstreams product as a whole. All labels in the UI can remain as is.

**Comment by andreivladbrg** [comment]
Interesting idea. I'm kinda pro this.

Between the suggested names, definitely the better version for us is `MerkleStreamer`, as it maintains the streaming "footprint" in the contract name. 

The only problem I can think of is that we will have different terminologies between the contracts and the app. e.g. Create airstream campaign -> create merkle streamer.

**Reply by PaulRBerg** [reply]
Thanks!

That's not a problem. >95% of all UI users will never care about the contract name. And among those who care, there's nothing wrong with using a different terminology in the UI ("airstreams").

Though we could also reconsider renaming "airstreams" to "Merkle streams" in the UI.

**Comment by razgraf** [comment]
Love it!

"Airstreams" are still quite cool branding-wise, so I belive we should keep them in the app and product materials. It sounds much more meme-able for consumers. But for integrators, having a more general purpose nomenclature helps us avoid narrowing down the spectrum of possibilities for that same code.

`MerkleStreamer` sounds cool. Using the more general `MerkleDistributor` might make people think the contract supports/is a regular ERC20 merkle distributor, whereas ours is a bit more specific.

**Reply by PaulRBerg** [reply]
Awesome, thanks.

Agree that we should keep using Airstreams in the UI.
